### PR TITLE
Reorganise and clean up the whole guide.

### PR DESCRIPTION
**Pull request by m-ou-se**
This reorganises the guide and removes a lot of stub sections.

### CODE REVIEW

File: src/feature-lifecycle/tracking-issues.md [hunk 1]
```diff
@@ -1,16 +0,0 @@
-# Using tracking issues
-
-**Status:** Stub
-
-Tracking issues are used to facilitate discussion and report on the status of standard library features. All public APIs need a dedicated tracking issue. Some larger internal units of work may also use them.
-
-## Creating a tracking issue
-
-There's a template that can be used to fill out the initial tracking issue. The Libs team also maintains [a Cargo tool](https://github.com/rust-lang/libs-team/tree/main/tools/unstable-api) that can be used to quickly dump the public API of an unstable feature.
```
Comment (jyn514) [inline src/feature-lifecycle/tracking-issues.md:9]: can you please link the `unstable-api` tool from the libs tracking issue template? https://github.com/rust-lang/rust/issues/new?template=library_tracking_issue.md

**Review by jyn514 — COMMENTED** [review]
this gets rid of a lot of the "broken windows", thanks :) reading the diff it does strike me that the "for reviewers" sections were acting as little summaries where you only have to read the bullet points rather than the whole section, but maybe people weren't using them? i don't feel strongly about keeping them around.

File: src/feature-lifecycle/tracking-issues.md (hunk 1 above)
Comment (m-ou-se) [reply src/feature-lifecycle/tracking-issues.md:9]: I'm pretty sure the tool is broken. It hasn't worked for a long time now. That's why I removed it.

File: src/feature-lifecycle/api-change-proposals.md
[the remark this answers is not in the document]
Comment (m-ou-se) [reply src/feature-lifecycle/api-change-proposals.md:1]: It's outdated and will be replaced by a new policy. See https://github.com/rust-lang/std-dev-guide/pull/55

File: src/SUMMARY.md [hunk 2]
```diff
@@ -1,53 +1,30 @@
 # Summary
 
-[About this guide](./about-this-guide.md)
-
-[Getting started](./getting-started.md)
+- [About this Guide](about.md)
+
+- [Contributing to the standard libraries]()
```
[the remark this answers is not in the document]
Comment (m-ou-se) [reply src/SUMMARY.md:5]: Yes, that's better than having a link that just goes to an empty filler page.

File: src/feature-lifecycle/tracking-issues.md (hunk 1 above)
Comment (jyn514) [reply src/feature-lifecycle/tracking-issues.md:9]: 😞 fair enough

File: src/tools-and-bots/bors.md [hunk 3]
```diff
@@ -1,15 +0,0 @@
...
-
-You can approve a PR with:
-
-```ignore
-@bors r+
-```
-
-## Rolling up
-
-For Libs PRs, rolling up is usually fine, in particular if it's only a new unstable addition or if it only touches docs. See the [rollup guidelines](https://forge.rust-lang.org/compiler/reviews.html#rollups) for more details on when to rollup.
```
[the remark this answers is not in the document]
Comment (m-ou-se) [reply src/tools-and-bots/bors.md:15]: Added it back to team/reviewing.

File: src/feature-lifecycle/deprecation.md
[the remark this answers is not in the document]
Comment (m-ou-se) [reply src/feature-lifecycle/deprecation.md:1]: There was almost nothing on this page. It says that APIs can't be removed from stable, but that doesn't need a separate page next to the 'breaking changes' page. The remaining info it had on *how* to handle a deprecation is outdated/incomplete. Once we properly figure out when and how to deprecate things, we should add that as a page under the new `policies/` section.

**Comment by m-ou-se** [comment]
@rust-lang/libs This reorganises and cleans up the std-dev-guide a bit. We had a lot of nearly empty pages and a structure that made it somewhat hard to just incrementally add new information. I removed all empty stubs and not-so-useful information and tried to categorize what remained into simpler categories. ("how to contribute", "policies", "team", etc.)

See https://std-dev-guide.rust-lang.org/

I'm not particularly tied to this structure/categorization; it's just what seemed to work well for the information we currently have. Feel free to move everything around if you think another structure works better. :)

I might have accidentally also removed some things I miscategorized as unuseful, as I tried to err a bit on the side of cleaning up too much rather than cleaning up too little, considering the amount of incomplete and outdated info there was. ^^

**Pull request merged**